PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:5 + add:0, change matches:
-    _inherit = ["ir.ui.view"]
-    _inherit = ['mail.message']
-    _inherit = ['res.config.settings']
-    _inherit = ['res.partner']
-    _inherit = ['res.users.apikeys.description']

Total Changes: 17

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/portal/models/ir_http.py
+++ b/addons/portal/models/ir_http.py
@@ -4,7 +4,7 @@ from odoo import models
 
 
 class IrHttp(models.AbstractModel):
-    _inherit = ['ir.http']
+    _inherit = 'ir.http'
 
     @classmethod
     def _get_translation_frontend_modules_name(cls):

--- a/addons/portal/models/ir_qweb.py
+++ b/addons/portal/models/ir_qweb.py
@@ -6,7 +6,7 @@ from odoo.tools import is_html_empty, lazy
 
 
 class IrQweb(models.AbstractModel):
-    _inherit = ["ir.qweb"]
+    _inherit = "ir.qweb"
 
     def _prepare_frontend_environment(self, values):
         """ Returns ir.qweb with context and update values with portal specific

--- a/addons/portal/models/ir_ui_view.py
+++ b/addons/portal/models/ir_ui_view.py
@@ -5,6 +5,6 @@ from odoo import models, fields
 
 
 class IrUiView(models.Model):
-    _inherit = ["ir.ui.view"]
+    _inherit = "ir.ui.view"
 
     customize_show = fields.Boolean("Show As Optional Inherit", default=False)

--- a/addons/portal/models/mail_message.py
+++ b/addons/portal/models/mail_message.py
@@ -7,7 +7,7 @@ from odoo.addons.portal.utils import get_portal_partner
 
 
 class MailMessage(models.Model):
-    _inherit = ['mail.message']
+    _inherit = 'mail.message'
 
     def portal_message_format(self, options=None):
         """ Simpler and portal-oriented version of 'message_format'. Purpose

--- a/addons/portal/models/mail_thread.py
+++ b/addons/portal/models/mail_thread.py
@@ -8,7 +8,7 @@ from odoo.addons.portal.utils import validate_thread_with_hash_pid, validate_thr
 
 
 class MailThread(models.AbstractModel):
-    _inherit = ['mail.thread']
+    _inherit = 'mail.thread'
 
     _mail_post_token_field = 'access_token' # token field for external posts, to be overridden
 

--- a/addons/portal/models/portal_mixin.py
+++ b/addons/portal/models/portal_mixin.py
@@ -7,6 +7,7 @@ from odoo import api, exceptions, fields, models, _
 
 
 class PortalMixin(models.AbstractModel):
+    _name = 'portal.mixin'
     _description = 'Portal Mixin'
 
     access_url = fields.Char(

--- a/addons/portal/models/res_config_settings.py
+++ b/addons/portal/models/res_config_settings.py
@@ -5,7 +5,7 @@ from odoo import api, fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     portal_allow_api_keys = fields.Boolean(
         string='Customer API Keys',

--- a/addons/portal/models/res_partner.py
+++ b/addons/portal/models/res_partner.py
@@ -5,7 +5,7 @@ from odoo import models
 
 
 class ResPartner(models.Model):
-    _inherit = ['res.partner']
+    _inherit = 'res.partner'
 
     def _can_edit_name(self):
         """ Name can be changed more often than the VAT """

--- a/addons/portal/models/res_users_apikeys_description.py
+++ b/addons/portal/models/res_users_apikeys_description.py
@@ -6,7 +6,7 @@ from odoo.exceptions import AccessError
 
 
 class ResUsersApikeysDescription(models.TransientModel):
-    _inherit = ['res.users.apikeys.description']
+    _inherit = 'res.users.apikeys.description'
 
     def check_access_make_key(self):
         try:
